tests/acceptance/features/apiSpaces/search.feature

```diff
@@ -121,3 +121,20 @@ Feature: Search
       | /SubFolder1/subFOLDER2/insideTheFolder.txt |
     But the search result of user "Alice" should not contain these entries:
       | /folderMain |
+
+
+  Scenario: search inside folder in shares
+    Given user "Alice" has created a share inside of space "find data" with settings:
+      | path      | folderMain |
+      | shareWith | Brian      |
+      | role      | viewer     |
+    And user "Brian" has accepted share "/folderMain" offered by user "Alice"
+    When user "Brian" searches for "folder" inside folder "/folderMain" in space "Shares" using the WebDAV API
+    Then the HTTP status code should be "207"
+    And the search result should contain "3" entries
+    And the search result of user "Brian" should contain only these entries:
+      | /SubFolder1                                |
+      | /SubFolder1/subFOLDER2                     |
+      | /SubFolder1/subFOLDER2/insideTheFolder.txt |
+    But the search result of user "Brian" should not contain these entries:
+      | /folderMain |
```

Comment:
maybe I misunderstood `And the search result of user "Brian" should contain only these entries:` step.
upto my knowledge this step fails if we put more or less resources in it so why do we need to check resources' existence in the new step? Thats already covered in `And the search result of user "Brian" should contain only these entries:`

Is there any specific reason for again checking resources not exist? IMO 
```suggestion

```

Reply:
seemes like the step `And the search result of user "Brian" should contain only these entries:` is not that strict so it's good to have this check here.

https://github.com/owncloud/ocis/pull/6987#discussion_r1287912557

Reply:
@amrita-shrestha you are right in a sense because that step checks the no of entries at first before matching the entries but i was suggested to double check using this But step in previous pr.
https://github.com/owncloud/ocis/pull/6949#discussion_r1282714882
CC: @saw-jan @PrajwolAmatya @SagarGi @grgprarup
suggestions appreciated

Reply:
I had seen the comment but my understanding this that we are going to refactor this step  `And the search result of user "Brian" should contain only these entries:` to avoid false positives (or false negatives). 
# Point of concern that revolve around my mind
- if we going to refactor `And the search result of user "Brian" should contain only these entries:`  and this step does strictly what it should do after the refactoring step. In this case `But the search result of user "Brian" should not contain these entries:` step only makes test time longer. 
- if we are not going to refactor `And the search result of user "Brian" should contain only these entries:` then  we can use this step `And the search result of user "Alice" should contain these entries:`

cc @SwikritiT @KarunAtreya 
IMO this two-step should not be used together 
```
 And the search result of user "Brian" should contain only these entries:
      | /SubFolder1                                |
      | /SubFolder1/subFOLDER2                     |
      | /SubFolder1/subFOLDER2/insideTheFolder.txt |
    But the search result of user "Brian" should not contain these entries:
      | /folderMain |
```

Reply:
@KarunAtreya This 2-step combination contains repeated checks like entries count double.
 While using step repetition should be minimal
```suggestion
    And the search result should contain "3" entries
    And the search result of user "Brian" should contain only these entries:
      | /SubFolder1                                |
      | /SubFolder1/subFOLDER2                     |
      | /SubFolder1/subFOLDER2/insideTheFolder.txt |
```

Reply:
```
 And the search result of user "Brian" should contain only these entries:
      | /SubFolder1                                |
      | /SubFolder1/subFOLDER2                     |
      | /SubFolder1/subFOLDER2/insideTheFolder.txt |
```
above step should be designed in such a way that we don't need to add this extra step
```
And the search result should contain "3" entries
    But the search result of user "Brian" should not contain these entries:
      | /folderMain |
```

Reply:
yes after `And the search result of user "Brian" should contain only these entries` is refactored to do what it is supposed to do we should not use both the steps together.